src/erlcloud_aws.erl

```diff
@@ -331,6 +337,20 @@ encode_params(Params, Headers) ->
     _ContentType -> {Params, LowerCaseHeaders}
   end.
 
+raw_xml_response(Body) ->
+    case format_xml_response(Body) of
+        {ok, XML} -> XML;
+        Error -> erlang:error(Error)
+    end.
+
+format_xml_response(Body) ->
+    try element(1, xmerl_scan:string(binary_to_list(Body))) of
+        XML -> {ok, XML}
+    catch
+        exit:{fatal,{expected_element_start_tag, _, _, _}} ->
+            {aws_error, {invalid_xml_response_document, Body}}
+    end.
```

Comment:
Thinking about this a little more, do we need to narrowly match errors? I am not sure about XML decoding, but I know JSON decoding can raise a variety of exceptions. I'd be fine to just blind-match them and treat them all as invalid XML document. Any reason not to?
```suggestion
format_xml_response(Body) ->
    XML = try 
        element(1, xmerl_scan:string(binary_to_list(Body)))
    catch
        _:_ ->
            {aws_error, {invalid_xml_response_document, Body}}
    end,
    {ok, XML}.
```

Reply:
I can think of a few bad reasons, but not any good ones.